### DISCUSSION TITLE
deploy Granian to Vercel for a Flask/Quart project

### DISCUSSION THREAD

**Discussion by rakshasha-medhi** [emmett-framework/granian#242] [Q&A] [answered]
Is there a way to use Granian when deploying a Flask/Quart project to Vercel?

In my local server test it works flawlessly but I hit a wall trying to use Granian with Vercel.
In my vercel.json I tried this:

```
{
  "builds": [
    {
      "src": "app.py",
      "use": "@vercel/granian --interface asgi app:app"
    }
  ],
  "routes": [
      {
        "src": "/(.*)",
        "dest": "app.py"
      }
  ],
  "regions": ["sin1"]
}
```

resulted in a build error:
npm ERR! enoent ENOENT: no such file or directory, open '/vercel/path0/.vercel/builders/@vercel/granian --interface asgi app:app/package.json'

In the builds - use I usually put @vercel/python for Flask/Quart projects and it works well.
Any help is appreciated, cheers!

**Comment by gi0baro** [comment]
I never used Vercel, but isn't a serverless solution? 
AFAIK the will have they're own *lambda function* which talks with your application using WSGI/ASGI interface, practically acting as a server, so in addition to the fact you shouldn't need a server in Vercel, even if you manage to get it running, you won't benefit from that at all, as it will be started for every request, instead of having that long running as it supposed to be..

**Reply by rakshasha-medhi** [reply, marked as answer]
Hi an update here: we can actually use Granian as a server runner in Vercel. one of my webapps has been using Granian and it works well.

Just need to add devCommand in the vercel.json file for example:
```
{
  "devCommand": "granian --interface asgi app:app",
  "version": 2,
  "builds": [
    {
      "src": "app.py",
      "use": "@vercel/python"
    }
  ],
  "routes": [
      {
        "src": "/(.*)",
        "dest": "app.py"
      }
  ]
}
```

Cheers!
